src/fsharp/FSharp.Build/Fsc.fs

```diff
@@ -178,15 +182,18 @@ type [<Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1704:Iden
     let mutable vslcid : string = null
     let mutable utf8output : bool = false
 
-#if ENABLE_MONO_SUPPORT
+    // See bug 6483; this makes parallel build faster, and is fine to set unconditionally
     // The property YieldDuringToolExecution is not available on Mono.
-    // So we only set it if available (to avoid a compile-time dependency). 
-    let runningOnMono = try System.Type.GetType("Mono.Runtime") <> null with e-> false         
-    do if not runningOnMono then  
-        typeof<ToolTask>.InvokeMember("YieldDuringToolExecution",(BindingFlags.Instance ||| BindingFlags.SetProperty ||| BindingFlags.Public),null,this,[| box true |])  |> ignore 
-#else
-    do this.YieldDuringToolExecution <- true  // See bug 6483; this makes parallel build faster, and is fine to set unconditionally
-#endif
+    // So we only set it if available (to avoid a compile-time dependency).
+    do
+        let runningOnMono = try System.Type.GetType("Mono.Runtime") <> null with e -> false
+        if not runningOnMono then  
+            typeof<ToolTask>.InvokeMember("YieldDuringToolExecution",
```

Comment:
Mono probably has this available these days in the DLL - if so ``do this.YieldDuringToolExecution <- true`` is much cleaner and long-term safer.  

If not I think I prefer the original code using ``#if``

Reply:
@dsyme xbuild is deprecated in favour of msbuild.  So yes I will remove the running on mono check.